cli/src/main.rs

```diff
@@ -1,8 +1,89 @@
+use db::PgPool;
+use tokio;
+use clap::{Parser, Subcommand};
+use db::sqlx;
+use std::fs;
+use std::io::Write;
+use uuid::Uuid; // Ensure you have the `uuid` crate in your Cargo.toml
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct KnowledgeArgs {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    Login,
+    Signup,
+    DisplayUsers,
+}
+
 #[tokio::main]
 async fn main() -> color_eyre::Result<()> {
-    let _db_pool = db::setup_db_pool().await?;
+    let args = KnowledgeArgs::parse();
+    let db_pool = db::setup_db_pool().await?;
+
+    match args.command {
+        Command::Login => {
+            println!("Login not implemented yet")
+        },
+        Command::Signup => {
+            sign_up(&db_pool).await?;
+        },
+        Command::DisplayUsers => {
+            display_users(&db_pool).await?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn persist_auth_session(user_id: Uuid) -> color_eyre::Result<()> {
+    if fs::metadata("auth.txt").is_ok() {
+        return Err(color_eyre::eyre::eyre!("auth.txt already exists"));
```

Comment:
Oh I think it would be kinda cool to have this error message tell you who you are currently logged in as! Either the `id` or even the username if its reasonable

What do you think?

Reply:
I think it's great idea, that way the user would know if they are logged in already or not